packages/connect-core/src/utils/misc.ts

```diff
@@ -1,3 +1,54 @@
 export function toArrayEntry<T>(value: T | T[]): T[] {
   return Array.isArray(value) ? value : [value]
 }
+
+type CacheStoreCallback<T> = () => Promise<T>
+type CacheStoreCached<T> = (
+  id: string,
+  callback: CacheStoreCallback<T>
+) => Promise<T>
+
+export function createCacheStore<T = string>(
+  limit: number = 10
+): {
+  cachedIndex: (id: string) => number
+  clear: () => void
+  get: (id: string, callback: CacheStoreCallback<T>) => Promise<T>
+  touch: (index: number) => void
+} {
+  const cachedEntries: [string, T][] = Array(limit)
```

Comment:
Given the type definition, the `cacheEntries` is an array of tuples with the first element a string and the second element of type T. That's why we use `entry[0]` on line 256 and we return the `cachedEntries[cachedIndex][1]`?

Reply:
Yes that’s correct!